drakrun/drakrun/apiscout.py

```diff
@@ -0,0 +1,116 @@
+import json
+import logging
+from operator import attrgetter
+from pathlib import Path, PureWindowsPath
+import pefile
+from typing import List, Dict, Any
+
+log = logging.getLogger(__name__)
+
+
+def get_bitness(pe: pefile.PE) -> int:
+    if pe.FILE_HEADER.Machine == pefile.MACHINE_TYPE["IMAGE_FILE_MACHINE_AMD64"]:
+        return 64
+    elif pe.FILE_HEADER.Machine == pefile.MACHINE_TYPE["IMAGE_FILE_MACHINE_I386"]:
+        return 32
+    else:
+        log.error(
+            f"Unsupported machine_type: {pe.FILE_HEADER.Machine} -> {pefile.MACHINE_TYPE[pe.FILE_HEADER.Machine]}"
+        )
+        return 0
+
+
+def get_product_version(pe: pefile.PE) -> str:
+    """
+    Based on https://stackoverflow.com/a/16076661/12452744
+    """
+
+    def LOWORD(dword):
+        return dword & 0x0000FFFF
+
+    def HIWORD(dword):
+        return dword >> 16
+
+    if len(pe.VS_FIXEDFILEINFO) != 1:
+        log.error("Unsupported case: len(pe.VS_FIXEDFILEINFO) != 1")
+        return "0.0.0.0"
+    try:
+        ms = pe.VS_FIXEDFILEINFO[0].ProductVersionMS
+        ls = pe.VS_FIXEDFILEINFO[0].ProductVersionLS
+        return "{}.{}.{}.{}".format(HIWORD(ms), LOWORD(ms), HIWORD(ls), LOWORD(ls))
+    except AttributeError:
+        log.exception("")
+        return "0.0.0.0"
```

Comment:
Why not simply return None and handle this case at the call site with `get_product_version() or "0.0.0.0"`